pylatex/document.py

```diff
@@ -221,16 +249,23 @@ def generate_pdf(self, filepath=None, *, clean=True, clean_tex=True,
                 ('pdflatex', [])
             )
 
-        main_arguments = ['--interaction=nonstopmode', basename + '.tex']
+        if python_cwd_available:
+            main_arguments = ['--interaction=nonstopmode', filepath + '.tex']
+        else:
+            main_arguments = ['--interaction=nonstopmode', basename + '.tex']
```

Comment:
The full `filepath` should work for non `cwd` available versions too. Can you change the PR to use this everywhere so the only difference is doing `chdir` or passing the `cwd` argument.

Reply:
If you mean to just rename the variable sure, I can probably do that, hopefully `filepath` is not used anywhere else down the line. But I guess if it works for cwd it should work for the old one as well.

If you mean that the content should be the same as well, than that's going to be more problematic. See I tried to be the least invasive, firstly because some design decisions I didn't completely understand and secondly I also have limited time to do community work. If I remember correctly I tried to make it work, that both of them use the absolute path at the end but for some reason one or more of the tests failed every time.